scripts/whitelist.js

```diff
@@ -0,0 +1,170 @@
+// The Carry token and the tokensale contracts
+// Copyright (C) 2018 Carry Protocol
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+const fs = require("fs");
+const CarryTokenPresale = artifacts.require("CarryTokenPresale");
+
+// Note that this script should be run through Truffle.  This requires
+// the environment variable WHITELIST_FILE which refers to the text file of
+// the line-separated addresses to be whitelisted.
+// 
+//     WHITELIST_FILE="..." \
+//     MNEMONIC="..." \
+//     npx truffle exec scripts/whitelist.js --network demo
+//
+// See also Truffle's docs:
+//
+//     https://truffleframework.com/docs/getting_started/scripts
+module.exports = callback => CarryTokenPresale.deployed().then(contract => {
+    const {
+        WHITELIST_FILE: filePath,
+        CHUNK: chunk = 260,
+        GAS_PRICE_GWEI: gasPriceGwei = "80",
+        OFFSET_ADDRESS: offsetAddress = null,
+    } = process.env;
+    if (!filePath) {
+        callback(new Error("Missing environment variable: WHITELIST_FILE"));
+        return;
+    }
+    if (typeof chunk != "number" && !chunk.match(/^\d+$/)) {
```

Comment:
When does `chunk` get converted to `number` if it is not a `number`?__

Reply:
It's one of JavaScript's some magic:

```js
> let chunk = "5";
undefined
> "foobar".slice(0, chunk);
'fooba'
```